### DISCUSSION TITLE
ARM iPXE boot issue on ThinkPad ARM laptop

### DISCUSSION THREAD

**Discussion by amdhugo** [ipxe/ipxe#1363] [General]
I compile a iPXE for ARM via this command from latest ipxe repo,
 _**make CROSS_COMPILE=aarch64-linux-gnu- ARCH=arm64 bin-arm64-efi/ipxe.efi EMBED=fix.ipxe -j 16**_
I deploy this efi as new name 'ipxe_arm64.efi' and update DHCP option to load correct efi file. Then, I use an ARM Laptop as ipxe client, however, I hit issue like this. 
![image](https://github.com/user-attachments/assets/a42ad64c-df5f-408e-bb1a-52ea9c4cd02f)

Is it issue on ipxe.efi? Anything missing in make file?

**Comment by NiKiZe** [comment]
It hangs before or while doing initial start of iPXE

If I'm not mistaken, you should not specify ARCH, check existing posts here for correct syntax, also try first with no local modifications and no embedded script, just so you get a working baseline.

**Reply by amdhugo** [reply]
If I don't specify ARCH, Does the new maked ipxe.efi support on ARM platform or both ARM and X86?

**Reply by NiKiZe** [reply]
Your build target is specified by `bin-arm64-efi/ipxe.efi`

**Reply by amdhugo** [reply]
Thanks NikuZe.
I made a new one without ARCH and will try it again tomorrow.

**Comment by mcb30** [comment]
> ![image](https://private-user-images.githubusercontent.com/64545075/396066986-a42ad64c-df5f-408e-bb1a-52ea9c4cd02f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNTkzNjQsIm5iZiI6MTczNDM1OTA2NCwicGF0aCI6Ii82NDU0NTA3NS8zOTYwNjY5ODYtYTQyYWQ2NGMtZGY1Zi00MDhlLWJiMWEtNTJlYTljNGNkMDJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE2VDE0MjQyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1YmI2M2ExNjc3OWVkOWYyOGY0MmViNmYyNzJlNDk0YTVhODk4Y2JkZjQ3YjhlODhjNjBiZmEzMjcwNDY3MGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.COeGO_kLvXTyxVK4Pm4WTiVQwgUHsMImzDm1Hpog07Q)
> 
> Is it issue on ipxe.efi? Anything missing in make file?

It looks as though your client is not even getting as far as downloading iPXE.  Check your TFTP server logs (or a packet capture) to see why the TFTP download is failing.

**Reply by amdhugo** [reply]
good suggestion, will do tomorrow.

**Comment by amdhugo** [comment]
Thank you mcb30 and NiKiZe. It works for me.
